hyp3_gamma/__main__.py

```diff
@@ -225,6 +224,9 @@ def water_map():
             f'--hand-threshold {args.hand_threshold} --hand-fraction {args.hand_fraction} '
             f'--membership-threshold {args.membership_threshold}', uselogging=True)
 
+    for p in Path(product_name).glob(f"{product_name}_WM_??_*.tif"):
+        p.unlink()
```

Comment:
I think it's worth explicitly deleting the four files we're targeting here instead of relying on `glob()`. It helps communicate the exact intent to the reader and reduces the risk of side effects if `water_map` adds additional outputs in the future:
```suggestion
    files_to_remove = [
        Path(product_name) / f'{product_name}_WM_VV_initial.tif',
        Path(product_name) / f'{product_name}_WM_VH_initial.tif',
        Path(product_name) / f'{product_name}_WM_VV_fuzzy.tif',
        Path(product_name) / f'{product_name}_WM_VV_fuzzy.tif',
    ]
    for file_to_remove in files_to_remove:
        file_to_remove.unlink()
```

Alternatively, we could tell `water_map` to place its output files in the root directory instead of the product directory, then explicitly move the files we want into the product directory. Or we could modify `water_map` to only output the fuzzy/initial files if requested, helping reduce overall disk usage.

Any thoughts? I think I'm content with the "run water map then delete the fuzzy/initial files" approach you have here.

Reply:
I like your idea to "run water map then delete the fuzzy/initial files". Deleting the explicit files is the better way from the point of view of the code developers.